### ISSUE TITLE
bucket_prefix parameter not taken into account

### ISSUE THREAD

**Issue by gregoryb** [jgroups-extras/jgroups-aws#318]
I start my server (Keycloak) with the following parameters 
`-Djgroups.s3.region_name=eu-west-1 -Djgroups.s3.bucket_name=mybucket -Djgroups.s3.bucket_prefix=keycloak/`

The ISPN directory is created at the root of the bucket.

I'm using `jgroups-aws-2.0.1.Final.jar`

**Comment by belaban** [comment]
Do you set `jgroups.s3.bucket_prefix` in the config, e.g.
`<S3_PING bucket_prefix="${jgroups.s3.bucket_prefix:ispn}"... />`?
I suspect that `bucket_prefix` is null. If you can debug this and set a breakpoint in `S3_PING.init()`, you'd know

**Comment by gregoryb** [comment]
I don't use config files. Only parameters with default values like in my initial post.

`ISPN` default directory is correctly created in `mybucket` by without the prefix.

Here is the log:
`using Amazon S3 ping in region eu-west-1 with bucket 'mybucket' and prefix ''`

I'm not a JAVA guy... complicated to debug this inside keycloak launch.

**Comment by belaban** [comment]
According to the log above, `bucket_prefix` is null, so it won't get created. Somewhere there is a config file (keycloak, infinispan), and that file must have your system props, or else setting system props is useless...

**Comment by rhusar** [comment]
> According to the log above, `bucket_prefix` is null, so it won't get created. Somewhere there is a config file (keycloak, infinispan), and that file must have your system props, or else setting system props is useless...

Correct, this project doesn't define these (e.g. jgroups.s3.bucket_name) - these must be defined in the stack.

@gregoryb Can you post what and how are you running?

**Comment by rhusar** [comment]
@belaban Should we perhaps consider making these configurable via environmental properties so as to 'standardize' the configuration across consumers?

**Comment by belaban** [comment]
@rhusar Yes, we can do this, but if Infinispan defines different system props for the same attributes, e.g. in their own config files, then this is moot.

**Comment by rhusar** [comment]
> > According to the log above, `bucket_prefix` is null, so it won't get created. Somewhere there is a config file (keycloak, infinispan), and that file must have your system props, or else setting system props is useless...
> 
> Correct, this project doesn't define these (e.g. jgroups.s3.bucket_name) - these must be defined in the stack.
> 
> @gregoryb Can you post what and how are you running?

@gregoryb Any update on this ^^^ ?

**Comment by gregoryb** [comment]
I described the issue on keycloak repository:
https://github.com/keycloak/keycloak/issues/26703

I don't know where is the best place for this as it's related to a Keycloak configuration.
I don't know what more I can do to help

**Comment by belaban** [comment]
@rhusar Great idea: +1!

**Comment by rhusar** [comment]
> [@rhusar](https://github.com/rhusar) Great idea: +1!

Implemented via https://github.com/jgroups-extras/jgroups-aws/pull/389, will be available in 3.0.1.Final. Closing.
